src/v/resource_mgmt/tests/cpu_profiler_test.cc

```diff
@@ -34,12 +48,27 @@ SEASTAR_THREAD_TEST_CASE(test_cpu_profiler) {
     // The profiler service will request samples from seastar every
     // 256ms since the sample rate is 2ms. So we need to be running
     // for at least that long to ensure the service pulls in samples.
-    auto end_time = ss::lowres_clock::now() + 256ms + 10ms;
-    while (ss::lowres_clock::now() < end_time) {
-        // yield to allow timer to trigger and lowres_clock to update
-        ss::thread::maybe_yield();
-    }
+    busy_loop(256ms + 10ms).get();
 
     auto results = cp.shard_results();
     BOOST_TEST(results.samples.size() >= 1);
 }
+
```

Comment:
Can you add 1 more test (or extend an existing test) which checks that nesting works, e.g., start a sample run with time 2P, then a second one with time P and when the latter completes check that the profiler is still enabled and that when the 2P run finishes it is disabled?

Then I think we'd have good coverage here.

Reply:
Added a nested override test.